Design-Docs/glossary.md

```diff
@@ -0,0 +1,143 @@
+---
+title: SCS Definitions (Draft)
+version: 2021-09-09-001
+authors: Matthias Hamm
+state: v1.0 (for R0)
+---
+
+Glossary - Naming conventions for SCS (Draft) 
+
+# Motivation
+
+In order to have a consistent naming scheme in SCS environments, we need a definition of the different terms. 
+
+# Proposal
+
+This is a draft version, which is free to be discussed
+
+## Definition of a Region
+
+An OpenStack/SCS Region consists of at least one or more Availability Zones that share a Control Plane with their services. As a result, they share an API, as well as a CEPH cluster, i.e. a data store. Within the region, any Layer 2 networks are available to the user.
+
+Regions can be federated when the SCS code is ready.
+
+A region is located in one fire compartment.
```

Comment:
I agree with Jonas and disagree with the proposal here.

For me, an "availability zone" is a failure domain. So you need to have a reasonable failure isolation between AZs of the same region, e.g. different rooms. My expectation of an AZ is that a broken switch, a fire, a power outage, a broken network uplink will very likely only take down one AZ (or none :-) ). There is some flexibility -- whether you have redundant internet connectivity shared between AZs or separate ones per AZ is up to you.

My definition would be:
A region consists of one or several availability zones, which are geographically located close to each other and are connected by low-latency (<2ms) high bandwidth (10s of Gbps) connections.

Reply:
Another question:
Don't we typically share identities across regions of one cloud provider? In OpenStack terms: the keystone is (logically) shared. (Whether we replicate the identities or have a central model with decentral caches is irrelevant.)
Federation is for me the next level, where I can authorize identities from several sources -- this does not seem to be required for a single provider multi region setup.
What do our IAM and OpenStack experts think here?

Reply:
All you said about identity management is correct as far as I know. Regions still trust one another regarding authentication and authorization, which may not be the case for federated deployments.

Reply:
>  A region consists of one or several availability zones, which are geographically located close to each other and are connected by low-latency (<2ms) high bandwidth (10s of Gbps) connections.

+unmetered

Also I guess we can bikeshed whether 2ms is really a good threshold. I am looking at a customer complaint at the moment where they have the issue that a latency of 1.5ms between database and app is noticeably slow in their setup.

Reply:
Low latency <2ms isn't necessarily possible in 2 DCs of the same region in case of fiber cut. You probably want redundant paths that aren't always optimal.

Reply:
I agree on the definition of AZ = failure domain, and I think that SCS community members should clarify what their failure domain means for them. It's a differentiator between vendors.

Reply:
To answer to @garloff , not all vendors implement federation or shared keystone. Some vendors prefer a shared nothing architecture... 

However, I believe in this case it would be wise to think about federation from day 1 for regions (even if it means k2k). AZs should not be a concern for keystone, as it should be a single keystone per region (with multiple load balanced endpoints). 
I am not personally fond of k2k but federation from day1 makes sense to me.